code/game/objects/items/cards_ids.dm

```diff
@@ -757,3 +757,30 @@ update_label("John Doe", "Clowny")
 
 /obj/item/card/id/job/lawyer
 	icon_state = "lawyer"
+
+/obj/item/card/id/pass
+	name = "promotion pass"
+	desc = "A card that, when swiped on your ID card, will grant you all the access. Should not substitute your actual ID card."
+	icon_state = "data_1"
+	registered_name = "Unregistered ID"
+	assignment = "Access Pass"
+
+/obj/item/card/id/pass/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(istype(target,/obj/item/card/id) && proximity)
+		var/obj/item/card/id/idcard = target
+		for(var/ac in access)
```

Comment:
Be descriptive with variable names, it doesn't cost anything extra

Reply:
A short iterator like this is fine, the name could be better, but I wouldn't list it as a cause for concern.